packages/auth/src/platform_browser/mfa/assertions/totp.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Stores the shared secret key and other parameters to generate time-based OTPs.
+ * Implements methods to retrieve the shared secret key, generate a QRCode URL.
+ */
+
+export class TotpSecret {
+  constructor(
+    readonly secretKey: string,
+    readonly hashingAlgorithm: string,
+    readonly codeLength: number,
+    readonly codeIntervalSeconds: number
+  ) {}
+  /**
+   * Returns the shared secret key/seed used to generate time-based one-time passwords.
+   *
+   * @returns Shared secret key/seed used for enrolling in TOTP MFA and generating otps.
+   */
+  sharedSecretKey(): string {
```

Comment:
Per the API discussion, let's get rid of these getters for now. @prameshj can fill in details offline